### PR TITLE
SlashCommand: Allow having groups and subcommands on the same level

### PR DESCRIPTION
**Pull request by Leo40Git**
this is legal now

Note that I haven't reconfigured the maximum subcommand/group limit (it still applies to subcommands and groups individually), and I can't actually find mention of this limit in the official documentation.

### CODE REVIEW

File: kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/commands/application/slash/_Functions.kt [hunk 1]
```diff
@@ -32,10 +32,6 @@ public suspend fun SlashCommand<*, *, *>.group(name: String, body: suspend Slash
         error("Command groups may not be nested inside subcommands.")
     }
 
-    if (subCommands.isNotEmpty()) {
```
Comment (gdude2002) [inline kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/commands/application/slash/_Functions.kt:35]: Isn't there also a set of subcommand functions that check whether there's a group? Please double-check that if you can, I don't have time atm

**Review by gdude2002 — CHANGES_REQUESTED** [review]

File: kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/commands/application/slash/_Functions.kt (hunk 1 above)
Comment (Leo40Git) [reply kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/commands/application/slash/_Functions.kt:35]: None of the `publicSubCommand` and `ephemeralSubCommand` functions in this file had a similar check (so theoretically, bots _could_ have already been using mixed groups/subcommands if they registered in the correct order).

Comment (gdude2002) [reply kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/commands/application/slash/_Functions.kt:35]: Oh, balls, okay. Well, thanks!

**Comment by gdude2002** [comment]
Thanks for that, good shout!

**Pull request merged**